reVX/handlers/sc_points.py

```diff
@@ -0,0 +1,618 @@
+            Resource gids associated with Supply curve point
+        gid_counts : list | str
+            Resource gid exclusion pixel counts
+        res_cf : ndarray
+            Resource gid capacity factor means
+        """
+        self._sc_gid = int(sc_gid)
+        self._capacity = capacity
+        res_order = np.argsort(res_cf)[::-1]
+        self._cf_means = res_cf[res_order]
+        self._res_gids = self._parse_list(res_gids)[res_order]
+        self._gid_counts = self._parse_list(gid_counts)[res_order]
+        self._res_capacity = \
+            self._gid_counts / np.sum(self._gid_counts) * self._capacity
+
+    def __repr__(self):
+        msg = "{} {}".format(self.__class__.__name__, self.sc_gid)
+        return msg
+
+    @property
+    def sc_gid(self):
+        """
+        Supply curve point gid
+
+        Returns
+        -------
+        int
+        """
+        return self._sc_gid
+
+    @property
+    def capacity(self):
+        """
+        Supply curve point capacity
+
+        Returns
+        -------
+        float
+        """
+        return self._capacity
+
+    @capacity.setter
+    def capacity(self, cap):
+        """
+        Update point's capacity
+
+        Parameters
+        ----------
+        cap : float
+        """
+        self._capacity = cap
+
+    @property
+    def resource_gids(self):
+        """
+        Resource gids associated with supply curve point
+
+        Returns
+        -------
+        ndarray
+        """
+        return self._res_gids
+
+    @property
+    def gid_counts(self):
+        """
+        Resource gid exclusion pixel counts
+
+        Returns
+        -------
+        ndarray
+        """
+        return self._gid_counts
+
+    @property
+    def resource_capacity(self):
+        """
+        Capacity of resource gids
+
+        Returns
+        -------
+        ndarray
+        """
+        return self._res_capacity
+
+    @property
+    def cf_means(self):
+        """
+        Capacity factor means for each resource gid
+
+        Returns
+        -------
+        ndarray
+        """
+        return self._cf_means
+
+    @staticmethod
+    def _parse_list(list_in, dtype=int):
+        """
+        Parse json list if needed
+
+        Parameters
+        ----------
+        list_in : str | list
+            List or jsonified list from supply curve table
+        dtype : np.dtype
+            dtype for output array
+
+        Returns
+        -------
+        out : ndarary
+            vector of values
+        """
+        if isinstance(list_in, str):
+            list_in = json.loads(list_in)
+        elif not isinstance(list_in, list):
+            msg = ('Must supply either a jsonified list or a list, not {}'
+                   .format(type(list_in)))
+            logger.error(msg)
+            raise ValueError(msg)
+
+        out = np.array(list_in, dtype=dtype)
+
+        return out
+
+    def _drop(self, build_capacity, drop=None):
```

Comment:
This is a confusing method... Consider renaming to build_capacity or drop_built_capacity. Additionally, the docstring is incomplete (add description of build_capacity parameter). 

On a technical note, this always builds out the first gids in the resource gid list right? What kind of guarantee is there that these are the gids that should actually be built? I think i see the sorting happening in __init__, but this should be described in the doc string here (build the resource gids with the best capacity factor)

Reply:
renamed and updated